### PR TITLE
Add api documentation

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**

Add [Open api 3](https://swagger.io/specification/) api docs in `docs/openapi`. Api documentation uses [Redoc](https://github.com/Redocly/redoc) to serve API documentation locally

**How to test**

You'll need to update dependencies (`yarn deps` or `yarn docker:deps`). Run `yarn docs:serve` (or `yarn docker:start`) and browse to `localhost:5000` to view the API documentation.

**Checklist**

~- [n/a] Meets issue criteria~
- [x] Code tested
~- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)~
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,5 @@ coverage
 .env
 junit.xml
 frontend/build
+openapi3.yaml
```
Comment (rahearn) [inline .gitignore:9]: What is this file? Should I have to generate it somehow?

File: README.md [hunk 2]
```diff
@@ -53,6 +55,7 @@ Other Commands
 | | Run the linter only for the backend | `yarn lint` | |
 | | Run the linter for the the backend with results output to xml files | `yarn lint:ci`| |
 | | Run `yarn lint:ci` for both the frontend and backend | `yarn lint:all`| |
+| | Host the open api 3 spec using [redoc](https://github.com/Redocly/redoc) at `localhost:5000` | `yarn docs:serve` | |
```
Comment (rahearn) [inline README.md:58]: No need to do this now (need to answer questions around authentication first anyway), but in the future I'd like to serve the api docs off of production as well. You don't foresee any issues using this framework to do that, do you?

**Review by rahearn — COMMENTED** [review]
Looks good. Please answer the couple inline questions before I merge this in.

File: docs/adr/0007-api-documentation.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Approved
+
+## Context
+
+There are a lot of tools available for documenting APIs. These tools generate easy to read documentation that can be used by frontend engineers and even third party developers. Setting up a documentation workflow will allow those engineers to integrate with the TTAHUB API faster and with more confidence. In addition, these API tools open a path for further testing and automation in the future, since the definition of the API is machine readable.
+
+## Decision
+
+We will document our API with [OpenAPI specification 3](https://swagger.io/specification/) (formally known as swagger).
```
[the remark this answers is not in the document]
Comment (jasalisbury) [reply docs/adr/0007-api-documentation.md:15]: :facepalm: formerly

File: .gitignore (hunk 1 above)
Comment (jasalisbury) [reply .gitignore:9]: I can remove that file from the gitignore. It used to be generated when we were using a different library to serve/generate the API documentation.

File: README.md (hunk 2 above)
Comment (jasalisbury) [reply README.md:58]: Should work fine, `redoc-cli` can generate a static HTML file that we can use in production (or any other deployed environment)

**Review by rahearn — APPROVED** [review]

**Pull request merged**